tests/python/gaia-ui-tests/gaiatest/mixins/treeherder.py

```diff
@@ -246,15 +246,12 @@ def post_to_treeherder(self, tests):
 
         # Send the collection to Treeherder
         url = urlparse(self.treeherder_url)
-        request = TreeherderRequest(
-            protocol=url.scheme,
-            host=url.hostname,
-            project=project,
-            oauth_key=os.environ.get('TREEHERDER_KEY'),
-            oauth_secret=os.environ.get('TREEHERDER_SECRET'))
+        client = TreeherderClient(protocol=url.scheme, host=url.hostname)
         self.logger.debug('Sending results to Treeherder: %s' %
                           job_collection.to_json())
-        response = request.post(job_collection)
+        response = client.post_collection(project,
```

Comment:
`post_collection` doesn't provide a return value anymore, I'd probably just suggest just calling it and letting it throw an exception if there's an error. Other than that, this looks fine. You'll also want to bump the required version of treeherder-client, of course.